query-engine/driver-adapters/src/proxy.rs

```diff
@@ -149,6 +154,11 @@ impl TransactionProxy {
         &self.options
     }
 
+    pub fn begin(&self) -> UnsafeFuture<impl Future<Output = quaint::Result<()>> + '_> {
+        self.closed.store(true, Ordering::Relaxed);
```

Comment:
this doesn't look right, `begin` doesn't close the transaction
```suggestion
```

Reply:
Agreed, I'll check this out.